docs/pipelines.md

```diff
@@ -516,6 +517,184 @@ In particular:
+
+```
+/tekton/scripts/script-0-zsg5v: 3: tasks.sum-inputs.results.sum-result: not found
+/tekton/scripts/script-0-2g2dr: 3: tasks.count-comments-before.results.count: not found
+```
+
+## Configuring `Pipeline` result with `finally`
+
+Final tasks can emit `Results` but results emitted from the final tasks can not be configured in the
+[Pipeline Results](#configuring-execution-results-at-the-pipeline-level). We are working on adding support for this.
+
+```yaml
+  results:
+    - name: comment-count-validate
+      value: $(finally.check-count.results.comment-count-validate)
+```
+
+In this example, `PipelineResults` is set to:
+
+```
+"pipelineResults": [
+  {
+    "name": "comment-count-validate",
+    "value": "$(finally.check-count.results.comment-count-validate)"
+  }
+],
+```
+
+### Configuring the Final Task execution order
+
+It's not possible to configure or modify the execution order of the final tasks. Unlike `Tasks` in a `Pipeline`,
+all final tasks run simultaneously and starts executing once all `PipelineTasks` under `tasks` have settled which means
+no `runAfter` can be specified in final tasks.
+
+### Specifying execution `Conditions` in Final Tasks
+
+`Tasks` in a `Pipeline` can be configured to run only if some conditions are satisfied using `conditions`. But the
+final tasks are guaranteed to be executed after all `PipelineTasks` therefore no `conditions` can be specified in
+final tasks.
+
+### `PipelineRun` Status with `finally`
+
+With `finally`, `PipelineRun` status does not change and is still calculated based on the result of `PipelineTask`
+under `tasks` section.
+
+Without `finally`, `PipelineRun` status is set to `true` and `reason` is set to `Succeeded` if all `PipelineTask` from
+`tasks` section are executed successfully. `PipelineRun` status is set to `true` and `reason` is set to `Completed` if
+one or more `PipelineTask` was skipped because of its own condition failure or was skipped because its parent
+was skipped. And `Pipeline` exits on the first failure of `PipelineTask` and `PipelineRun` status is set to `false` and
+`reason` is set to `failed`.
+
+With `finally`, `PipelineRun` status is still set to `false` if any of the `PipelineTask` from `tasks` section fails
+irrespective of any final task failure.
+
+```none
+apis.Condition{
+  Type:    apis.ConditionSucceeded,
+  Status:  corev1.ConditionFalse,
+  Reason:  ReasonFailed,
+  Message: fmt.Sprintf("TaskRun %s has failed", tasks.TaskRun.Name),
+}
+```
+
+And `PipelineRun` status is set to `true` if all of the `PipelineTask` from `tasks` section succeed irrespective of
+any final task failure.
+
+```none
+apis.Condition{
+  Type:    apis.ConditionSucceeded,
+  Status:  corev1.ConditionTrue,
+  Reason:  Succeeded, # or Completed
+  Message: fmt.Sprintf("Tasks Completed: %d, Skipped: %d", len(tasks)+len(finally), skipTasks),
+}
+```
+
+Note that it could be possible that one or all final tasks fail but `PipelineRun` status does not reflect
```

Comment:
I am not sure about this. If I have a pipeline that create a cluster (`openshift-install`), does things, and have a finally to remove delete the cluster. If the whole pipeline succeeded **but** the finally fails, I would consider this as a failure (even though it's at the cleaning part) because this means I may have a *dangling* cluster running around.

Reply:
or I misunderstood the status part. I feel a table showing the different state and result would be nice :angel:

Reply:
replaced the explanation with table, [here](https://github.com/tektoncd/pipeline/blob/634777213080a469b254cfb2c315a8174e997549/docs/pipelines.md#pipelinerun-status-with-finally)